### PR TITLE
Cut v2.10.0-rc.0

### PR DESCRIPTION
**Pull request by beorn7**
Draft for the changelog changes for early review before cutting the release tomorrow. (Feel free to review even if you are not on the reviewers list. I just picked people from -team that showed up often in the Git log.)

### CODE REVIEW

**Comment by beorn7** [comment]
Please also let me know if you see any red flags for the RC tomorrow, or you have things in flight that really should be part of the release.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+* [ENHANCEMENT] TSDB: Simplify mergedPostings.Seek, resulting in better performance if there are many posting lists. tsdb#595
+* [ENHANCEMENT] Cmd/promtool: Use POST requests for Query and QueryRange. client_golang#557
+* [ENHANCEMENT] Web: Fixed Content-Type for js and css instead of using `/etc/mime.types`. #5551
+* [ENHANCEMENT] Web: Update to Bootstrap 4.3.1. #5506
+* [BUGFIX] TSDB: Don't panic and recover nicely when running out of disk space. tsdb#582
+* [BUGFIX] TSDB: Correctly handle empty labels. tsdb#594
+* [BUGFIX] TSDB: Don't crash on an unknown tombstone reference. tsdb#604
+* [BUGFIX] Storage/remote: Remove queue-manager specific metrics if queue no longer exists. #5445 #5485 #5555
+* [BUGFIX] PromQL: Correctly display `{__name__="a"}`. #5552
+* [BUGFIX] Discovery/kubernetes: Use `service` rather than `ingreass` as the name for the service workqueue. #5520
```
Comment (simonpasquier) [inline CHANGELOG.md]: s/ingreass/ingress/

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+* [FEATURE] Web UI: Sort alerts by group name. #5448
+* [FEATURE] TSDB: Add `prometheus_tsdb_wal_segment_current` metric for the WAL segment index that TSDB is currently writing to. tsdb#601
+* [FEATURE] Scrape: Add `scrape_series_added` per-scrape metric. #5546
+* [FEATURE] Log filesystem type on startup. #5558
+* [ENHANCEMENT] Discovery/kubernetes: Add labels `__meta_kubernetes_endpoint_node_name` and `__meta_kubernetes_endpoint_hostname`. #5571
+* [ENHANCEMENT] Discovery/azure: Add label `__meta_azure_machine_public_ip`. #5475
+* [ENHANCEMENT] TSDB: Simplify mergedPostings.Seek, resulting in better performance if there are many posting lists. tsdb#595
+* [ENHANCEMENT] Cmd/promtool: Use POST requests for Query and QueryRange. client_golang#557
+* [ENHANCEMENT] Web: Fixed Content-Type for js and css instead of using `/etc/mime.types`. #5551
+* [ENHANCEMENT] Web: Update to Bootstrap 4.3.1. #5506
```
Comment (simonpasquier) [inline CHANGELOG.md]: Not sure if we need to list this as there's no user-facing impact.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,23 @@
+## 2.10.0 / 2019-05-22
+
+* [FEATURE] Template expansion: Make external labels available as `$externalLabels` in alert and console template expansion. #5463
+* [FEATURE] Web UI: Sort alerts by group name. #5448
```
Comment (brian-brazil) [inline CHANGELOG.md]: ENHANCEMENT, UI tweaks aren't that big.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,23 @@
...
+* [FEATURE] TSDB: Add `prometheus_tsdb_wal_segment_current` metric for the WAL segment index that TSDB is currently writing to. tsdb#601
+* [FEATURE] Scrape: Add `scrape_series_added` per-scrape metric. #5546
+* [FEATURE] Log filesystem type on startup. #5558
```
Comment (brian-brazil) [inline CHANGELOG.md]: ENHANCEMENT, and I'm not sure I'd list it as it's mostly for our own support purposes.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,23 @@
...
+* [ENHANCEMENT] Discovery/kubernetes: Add labels `__meta_kubernetes_endpoint_node_name` and `__meta_kubernetes_endpoint_hostname`. #5571
+* [ENHANCEMENT] Discovery/azure: Add label `__meta_azure_machine_public_ip`. #5475
+* [ENHANCEMENT] TSDB: Simplify mergedPostings.Seek, resulting in better performance if there are many posting lists. tsdb#595
+* [ENHANCEMENT] Cmd/promtool: Use POST requests for Query and QueryRange. client_golang#557
+* [ENHANCEMENT] Web: Fixed Content-Type for js and css instead of using `/etc/mime.types`. #5551
```
Comment (brian-brazil) [inline CHANGELOG.md]: Hmm, bugfix I'd say

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,23 @@
...
+* [ENHANCEMENT] Web: Update to Bootstrap 4.3.1. #5506
+* [BUGFIX] TSDB: Don't panic and recover nicely when running out of disk space. tsdb#582
```
Comment (brian-brazil) [inline CHANGELOG.md]: I'd put a comma after panic.

File: CHANGELOG.md (hunk 2 above)
Comment (beorn7) [reply CHANGELOG.md]: I don't feel strongly. It looked to me that the UI was visually a bit different, and that might make people wondered what happened. But my impression might have been wrong anyway. Does anybody know if the Bootstrap update changed the visual appearance slightly?

File: CHANGELOG.md (hunk 3 above)
Comment (beorn7) [reply CHANGELOG.md]: Fine with me.

File: CHANGELOG.md (hunk 4 above)
Comment (beorn7) [reply CHANGELOG.md]: I'd still list it. Happy to demote it to ENHANCEMENT.

File: CHANGELOG.md (hunk 5 above)
Comment (beorn7) [reply CHANGELOG.md]: Whatever. Will turn it into BUGFIX

File: CHANGELOG.md (hunk 6 above)
Comment (beorn7) [reply CHANGELOG.md]: Yes, much better.

Comment (beorn7) [reply CHANGELOG.md]: Hmm, perhaps better reword more. I changed it to "Don't panic when running out of disk space and recover nicely from the condition."

**Comment by beorn7** [comment]
Updated with #5587 and #5582.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,25 @@
+## 2.10.0 / 2019-05-22
+
+* [CHANGE/BUGFIX] API: Encode alert values as string to correctly represent Inf/NaN. #5582
+* [FEATURE] Template expansion: Make external labels available as `$externalLabels` in alert and console template expansion. #5463
```
Comment (brian-brazil) [inline CHANGELOG.md]: Didn't we also add $params etc. to console templates also?

File: CHANGELOG.md (hunk 2 above)
Comment (simonpasquier) [reply CHANGELOG.md]: I didn't notice any visible change but I didn't look very carefully either.

File: CHANGELOG.md (hunk 7 above)
Comment (beorn7) [reply CHANGELOG.md]: Yes, good catch. I'll add it as as ENHANCEMENT.

**Comment by beorn7** [comment]
Adjusted the header from 2.10.0 to 2.10.0-rc.0

File: CHANGELOG.md (hunk 2 above)
Comment (beorn7) [reply CHANGELOG.md]: To get to the bottom of this, I did a side-by-side comparison. There is somehow a one-pixel move, but no real change in appearance. (That happened at an earlier version.)

I'll remove this line from the changelog.

**Comment by beorn7** [comment]
Screenshot_20190522_150519

Look at the lack of gaps in the new release. The benchmark Prom is quite heavy, thus I guess we see the effects of https://github.com/prometheus/client_golang/pull/568 .

**Comment by simonpasquier** [comment]
Yes! :tada:

**Comment by krasi-georgiev** [comment]
do you still need the test running?

**Comment by beorn7** [comment]
https://github.com/prometheus/prometheus/blob/master/RELEASE.md#release-shepherd-responsibilities says: “With the pre-release the release shepherd is responsible for running and monitoring a benchmark run of the pre-release for 3 days, after which, if successful, the pre-release is promoted to a stable release.”

I guess now we have to release with only close to 2d of testing. Starting the benchmark again will most likely tell us nothing new. I assume the 3d are in the docs because sometimes problems only show up after that time.

In general, I'd prefer to let the release shepherd do their thing and only intervene if they seem to have dropped the ball.

**Comment by simonpasquier** [comment]
@beorn7 very sorry for having jumped too fast here.

**Comment by krasi-georgiev** [comment]
I think the 3 days was picket randomly and not for a specific technical reason so no problem if it is canceled a bit early if the tests look normal.

**Comment by beorn7** [comment]
@krasi-georgiev my conclusion would be partially the opposite: The test can be canceled early if it looks weird (because then we already know something has to be fixed). If it looks fine for 2d, however, we still don't know if there is perhaps an issue that only shows up after 3d.

The other part is the question if 3d is too long to be reasonable. I guess the best way to find out what others think is to open a PR, changing it in RELEASE.md, and ask others for their input.

**Pull request merged**